_inc/client/components/reconnect-modal/test/component.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+import sinon from 'sinon';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { ReconnectModal } from '../index';
+
+describe( 'ReconnectModal', () => {
+
+	let testProps = {
+		show:               true,
+		onHide:             noop,
+		isSiteConnected:    true,
+		isReconnectingSite: false,
+	};
+
+	describe( 'Initially', () => {
+
+		const wrapper = shallow( <ReconnectModal { ...testProps } /> );
+
+		it( 'renders the modal', () => {
+			expect( wrapper.find( 'Modal' ) ).to.have.length( 1 );
+		} );
+
+		it( 'has a Cancel button', () => {
+			expect( wrapper.find( '.reconnect__modal-cancel' ) ).to.have.length( 1 );
+		} );
+
+		it( 'its text is: Cancel', () => {
+			expect( wrapper.find( '.reconnect__modal-cancel' ).first().render().text() ).to.be.equal( 'Cancel' );
+		} );
+
+		it( 'has an onClick method', () => {
+			expect( wrapper.find( '.reconnect__modal-cancel' ).first().props().onClick ).to.exist; 
+		} );
+
+		it( 'when clicked, closeModal() is called once', () => {
+			const currentTestProps = {
+				show:               true,
+				isSiteConnected:    true,
+				isReconnectingSite: false,
+			};
+
+			const closeModal = sinon.spy();
+
+			class ReconnectModalMock extends ReconnectModal {
+				constructor( props ) {
+					super( props );
+					this.closeModal = closeModal;
+				}
+			}
+			// We need to set the testProps again here, to make sure they are not affected by
+			// other tests running in between.
+			Object.assign( testProps, currentTestProps );
+			const wrapper = shallow( <ReconnectModalMock { ...testProps } /> );
+
+			wrapper.find( '.reconnect__modal-cancel'  ).simulate('click', { preventDefault: () => undefined });
+			expect( closeModal.calledOnce ).to.be.true;
+		} );
+
+		it( 'has a Reconnect button', () => {
+			expect( wrapper.find( '.reconnect__modal-reconnect' ) ).to.have.length( 1 );
+		} );
+
+		it( 'its text is: Reconnect Jetpack', () => {
+			expect( wrapper.find( '.reconnect__modal-reconnect' ).first().render().text() ).to.be.equal( 'Reconnect Jetpack' );
+		} );
+
+		it( 'has an onClick method', () => {
+			expect( wrapper.find( '.reconnect__modal-reconnect' ).first().props().onClick ).to.exist; 
+		} );
+
+		it( 'when clicked, clickReconnectSite() is called once', () => {
+			const currentTestProps = {
+				show:               true,
+				isSiteConnected:    true,
+				isReconnectingSite: false,
+			};
+
+			const clickReconnectSite = sinon.spy();
+
+			class ReconnectModalMock extends ReconnectModal {
+				constructor( props ) {
+					super( props );
+					this.clickReconnectSite = clickReconnectSite;
+				}
+			}
+			// We need to set the testProps again here, to make sure they are not affected by
+			// other tests running in between.
+			Object.assign( testProps, currentTestProps );
```

Comment:
Would using a `before` statement help here? If we don't just set the `testProps` once, but do it instead in a `before` function, they would be the same as the initial ones.

Reply:
@zinigor I have added the `before` approach, but in order to get the intended test properties, aka inside every describe: defaults + **only** current block changes, I had to also add nested `before` blocks.
Let me know if that approach works or if there's a better alternative.
Many thanks for reviewing this PR!

Reply:
Thanks! This is exactly the approach that I had in mind. This way we're clearly separating test data preparation from other `describe` blocks, so we don't have any side effects from other tests.